lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -395,10 +394,11 @@ namespace utf8
             cch -= 4;
         }
 LSlowPath:
+        // WARNING: Assume cch correct, suppress end-of-buffer checking
+        // we might have failed to measure cch, +1 for odd length. sizeof(char16) == 2
+        LPCUTF8 end = ptr + cch + (cch % 2 ? 2 : 1);
         while (cch-- > 0)
         {
-            LPCUTF8 end = ptr + cch + 1; // WARNING: Assume cch correct, suppress end-of-buffer checking
```

Comment:
It depends on how this function was supposed to be used. The comment said assuming passed in "cch" is known and correct. This function might intend to be used in scenarios that we've already decoded the stream before and know for sure the character counts. @boingoing may know more context.